File: EndGate/EndGate.Core.Net/Loopers/TimedCallback.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EndGate.Core.Net.Loopers
{
    internal class TimedCallback : LooperCallback
    {
        private int _fps;

        public TimedCallback()
        {
        }

        public TimedCallback(int fps, Action callback) :
            base(callback)
        {
            Fps = fps;
        }

        public int TriggerFrequency { get; private set; }
        public long LastTriggered { get; set; }
        public int Fps
        {
            get
            {
                return _fps;
            }
            set
            {
                _fps = value;

                TriggerFrequency = (_fps != 0) ? 1000 / _fps : 0;
            }
        }
    }
}

File: EndGate/EndGate.Core.JS/Loopers/TimedCallback.ts
/// <reference path="LooperCallback.ts" />
/// <reference path="../Interfaces/ITyped.ts" />

module EndGate._.Loopers {

    export class TimedCallback extends LooperCallback implements ITyped {
        public _type: string = "TimedCallback";

        constructor(fps: number, callback: Function) {
            super(callback);

            this.Fps = fps;
            this.TimeoutID = 0;
            this.Active = false;
        }

        public Fps: number;
        public TimeoutID: number;
        public Active: bool;
    }

}